File: app/src/main/java/org/deiverbum/app/core/module/DatabaseModule.java
package org.deiverbum.app.core.module;

import android.app.Application;

import androidx.room.Room;

import org.deiverbum.app.data.db.AppDatabase;
import org.deiverbum.app.data.db.dao.TodayDao;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import dagger.hilt.InstallIn;
import dagger.hilt.components.SingletonComponent;

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Module
@InstallIn(SingletonComponent.class)

public class DatabaseModule {

    @Provides
    @Singleton
    public static AppDatabase provideDatabase(Application application) {
        return Room.databaseBuilder(application, AppDatabase.class,
                        "LiturgiaPlusDB")
                .createFromAsset("database/liturgia_202301000.db")
                .fallbackToDestructiveMigration()
                .allowMainThreadQueries()
                .build();
    }

    @Provides
    @Singleton
    public static TodayDao provideTodayDao(AppDatabase appDB) {
        return appDB.todayDao();
    }
}

File: app/src/main/java/org/deiverbum/app/core/module/DatabaseModule.kt
package org.deiverbum.app.core.module

import android.app.Application
import androidx.room.Room.databaseBuilder
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent
import org.deiverbum.app.data.db.AppDatabase
import javax.inject.Singleton

/**
 * @author A. Cedano
 * @version 1.0
 * @since 2023.1
 */
@Module
@InstallIn(SingletonComponent::class)
object DatabaseModule {
    @JvmStatic
    @Provides
    @Singleton
    fun provideDatabase(application: Application?): AppDatabase {
        return databaseBuilder(
            application!!, AppDatabase::class.java,
            "LiturgiaPlusDB"
        )
            .createFromAsset("database/liturgia_202301000.db")
            .fallbackToDestructiveMigration()
            //.allowMainThreadQueries()
            .build()
    }

    @JvmStatic
    @Provides
    @Singleton
    fun provideTodayDao(appDB: AppDatabase): org.deiverbum.app.data.db.dao.TodayDao {
        return appDB.todayDao()
    }

    @Provides
    @Singleton
    fun providePrayDao(appDB: AppDatabase): org.deiverbum.app.data.database.dao.TodayDao {
        return appDB.prayDao()
    }
}